### PR TITLE
Add examples to methods on integer types

### PR DESCRIPTION
**Pull request by ghost**
Adds `Example` sections to the rest of the integer methods.

cc @steveklabnik

### CODE REVIEW

**Comment by alexcrichton** [comment]
@bors: r+ 4a31f3ddf9120a5758e22ad714fc688c89be4902

Thanks!

**Comment by steveklabnik** [comment]
@bors: r-

this failed on travis: https://s3.amazonaws.com/archive.travis-ci.org/jobs/90055835/log.txt

plus i have some inline comments

File: src/libcore/num/mod.rs [hunk 1]
```diff
@@ -416,6 +416,13 @@ macro_rules! int_impl {
 
         /// Saturating integer addition. Computes `self + other`, saturating at
         /// the numeric bounds instead of overflowing.
+        ///
+        /// # Examples
+        ///
+        /// ```rust
```
Comment (steveklabnik) [inline src/libcore/num/mod.rs]: we don't use the `rust` in API documentation, can these be removed?

File: src/libcore/num/mod.rs [hunk 2]
```diff
@@ -416,6 +416,13 @@ macro_rules! int_impl {
 
         /// Saturating integer addition. Computes `self + other`, saturating at
         /// the numeric bounds instead of overflowing.
+        ///
+        /// # Examples
+        ///
```
Comment (steveklabnik) [inline src/libcore/num/mod.rs]: I've been adding `Basic usage:` before the first example block, can you do that as well, for consistency?

File: src/libcore/num/mod.rs [hunk 3]
```diff
@@ -1147,6 +1295,13 @@ macro_rules! uint_impl {
         /// negative minimal value for the type); this is a positive
         /// value that is too large to represent in the type. In such
         /// a case, this function returns `MIN` itself.
+        ///
+        /// # Examples
+        ///
+        /// ```rust
+        /// assert_eq!(100i8.wrapping_neg(), -100);
```
Comment (steveklabnik) [inline src/libcore/num/mod.rs]: Travis is showing this as failing: 

	<anon>:2:16: 2:20 error: invalid suffix `i` for numeric literal
<anon>:2     assert_eq!(100i 8.wrapping_neg(), -100);
                        ^~~~
<anon>:2:16: 2:20 help: the suffix must be one of the integral types (`u32`, `isize`, etc)
<anon>:2:21: 2:22 error: no rules expected the token `8`
<anon>:2     assert_eq!(100i 8.wrapping_neg(), -100);
                             ^

I'm not sure why. Looks like maybe you need parens? Seems... odd.

Comment (Toby-S) [inline src/libcore/num/mod.rs]: I think that's the old travis build; I pushed a second commit shortly after, which [built successfully](https://travis-ci.org/rust-lang/rust/builds/90069616). Will follow up on the other comments

**Comment by Toby-S** [comment]
@steveklabnik I've addressed the inline comments :-)

**Comment by steveklabnik** [comment]
Great! Let's see what Travis says, if it passes, r=me

File: src/libcore/num/mod.rs [hunk 4]
```diff
@@ -140,6 +140,7 @@ macro_rules! int_impl {
         /// Leading and trailing whitespace represent an error.
         ///
         /// # Examples
+        /// Basic usage:
```
Comment (steveklabnik) [inline src/libcore/num/mod.rs]: Oh, sorry, can you put a space between this and the `# Examples`? This is cramped, I know it still works markdown-wise

**Comment by steveklabnik** [comment]
After travis passes and that one last nit, r=me

(Sorry for being so nit-picky, consistent style is hard)

**Comment by Toby-S** [comment]
@steveklabnik No problem at all; looks like Travis has passed

**Comment by Toby-S** [comment]
r? steveklabnik

**Comment by steveklabnik** [comment]
:confetti_ball: 

thanks so much again

**Pull request merged**